crates/bounded_planet_core/src/bin/server/main.rs

```diff
@@ -123,9 +121,9 @@ pub struct PongLoggerState {
 fn log_pongs(mut state: ResMut<PongLoggerState>, receiver: ResMut<Events<ReceiveEvent>>) {
     for evt in state.event_reader.iter(&receiver) {
         if let ReceiveEvent::ReceivedPacket { data, .. } = evt {
-            if let Packet::Pong(Pong { timestamp }) = **data {
+            if let Packet::Pong(ref pong) = *data {
```

Comment:
I'm pretty sure this doesn't actually need to be fancy and use the ref and stuff...? Like with the others, it should be able to have the deref removed and it should all work out.